### PR TITLE
ported to OS X

### PR DESCRIPTION
**Pull request by atriwidada**
small modification to work around lack of clock_gettime in os x
adopted from https://gist.github.com/jbenet/1087739

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -201,8 +201,6 @@ AC_SUBST(LIBOPING_PC_LIBS_PRIVATE)
 
 AC_SEARCH_LIBS([nanosleep],[rt],[],
 		[AC_MSG_ERROR([cannot find nanosleep])])
-AC_SEARCH_LIBS([clock_gettime],[rt],[],
```
Comment (octo) [inline configure.ac]: I think the `AC_SEARCH_LIBS()` macro is still required for platforms where `clock_gettime()` is provided by the `rt` library (Solaris, I think). Just change the "action if not found" from `[AC_MSG_ERROR(…)]` to `[]`.

File: src/oping.c [hunk 2]
```diff
@@ -1615,8 +1620,20 @@ static void update_host_hook (pingobj_iter_t *iter, /* {{{ */
...
 	{
 		struct timespec ts = { 0, 0 };
-
-		if (clock_gettime (CLOCK_REALTIME, &ts) == 0)
+		int cret;
+
+#ifdef __MACH__ // OS X does not have clock_gettime, use clock_get_time
+		clock_serv_t cclock;
+		mach_timespec_t mts;
+		host_get_clock_service (mach_host_self (), CALENDAR_CLOCK, &cclock);
```
Comment (octo) [inline src/oping.c]: At your option: You're allocating and deallocating the clock on every response. Could the `clock_serv_t` object be longer lived?

File: src/oping.c [hunk 3]
```diff
@@ -1615,8 +1620,20 @@ static void update_host_hook (pingobj_iter_t *iter, /* {{{ */
...
+		cret = clock_get_time (cclock, &mts);
+		mach_port_deallocate (mach_task_self (), cclock);
+		ts.tv_sec = mts.tv_sec;
```
Comment (octo) [inline src/oping.c]: Nit, at your option:

```c
ts = (struct timespec) {
  .tv_sec = mts.tv_sec,
  .tv_nsec = mts.tv_nsec,
};
```

**Comment by octo** [comment]
I'm terribly sorry, but I lost track of this and have just fixed #19 (the same issue) myself. Is there any benefit in using this Mac OS X specific solution over using `gettimeofday()`? If so, please re-open!

Best regards,
—octo

**Comment by atriwidada** [comment]
Hi Octo,

I prefer a more portable approach. Since you feel that gettimeofday() can be used here, I think we don't need to use OS X specific handling anymore.

Regards.
andika

**Comment by octo** [comment]
Awesome, thanks @atriwidada!

**Pull request closed without merging**